js/post/update_post.js

```diff
@@ -316,74 +312,97 @@ function fill400s() {
                 maj_extra[k].value = '';
+                maj_extra[k].readOnly = false;
+            }
+        } 
+
+        // add credit count for MAT and STA courses
+        if (spec_extra[k].value.indexOf('MAT') > -1 || spec_extra[k].value.indexOf('STA') > -1) {
+            creditCount300and400['spec'] += 0.5;
+            filled_textboxes_extra['spec'] += 1;
+        } if (k < 3) {
+            if (maj_extra[k].value.indexOf('MAT') > -1 || maj_extra[k].value.indexOf('STA') > -1) {
+                creditCount300and400['maj'] += 0.5;
+                filled_textboxes_extra['maj'] += 1;
             }
         }
     }
 
     // fill courses that have been selected
-    for (m = 0; m < active300s.length; m++) {
-        if ((index300 === active300s.length) || (i === 4)) {
+    for (var m = 0; m < active300s.length; m++) {
+        if ((index300['spec'] === active300s.length) || (i === 4)) {
             break;
         }
         if ((i < 2) && (maj_extra[i].value === '')) {
-            maj_extra[i].value = active300s[index300];
+            maj_extra[i].value = active300s[index300['maj']];
+            maj_extra[i].readOnly = true;
+            index300['maj'] += 1;
+            filled_textboxes_extra['maj'] += 1;
+            creditCount300and400['maj'] += 0.5;
         }
         if (spec_extra[i].value === '') {
-            spec_extra[i].value = active300s[index300];
-            index300 += 1;
+            spec_extra[i].value = active300s[index300['spec']];
+            spec_extra[i].readOnly = true;
+            index300['spec'] += 1;
+            filled_textboxes_extra['spec'] += 1;
+            creditCount300and400['spec'] += 0.5;
         } 
         i += 1;
     }
 
     if (i < 4) {
-        for (m = 0; m < active400s.length; m++) {
-            if ((index400 === active400s.length) || (i === 4)) {
+        for (var m = 0; m < active400s.length; m++) {
+            if (((index400['spec'] === active400s.length) && (index400['maj'] === active400s.length)) || (i === 4)) {
                 break;
             }
-            if ((i < 2) && (maj_extra[i].value === '')) {
-                maj_extra[i].value = active400s[index400];
+            if ((i < 3) && (maj_extra[i].value === '')) {
+                maj_extra[i].value = active400s[index400['maj']];
+                maj_extra[i].readOnly = true;
+                index400['maj'] += 1;
+                filled_textboxes_extra['maj'] += 1;
+                creditCount300and400['maj'] += 0.5;
             }
-            if (spec_extra[i].value === '') {
-                spec_extra[i].value = active400s[index400];
-                index400 += 1;
+            if ((spec_extra[i].value === '') && (index400['spec'] < active400s.length)){
+                spec_extra[i].value = active400s[index400['spec']];
+                spec_extra[i].readOnly = true;
+                index400['spec'] += 1;
+                filled_textboxes_extra['spec'] += 1;
```

Comment:
Can you update these to dot notation? Much of the other JS code uses dot notation, though it hasn't been added to the style guide yet.